File: src-tauri/implementations/python/3/factorial/wrapper.py
import time
import sys
from memory_profiler import memory_usage
from factorial import factorial
from six import print_

def main():
    start_time = time.time()
    mem_usage = memory_usage((factorial, (8,)), max_usage=True)
    result = factorial(8)
    end_time = time.time()
    elapsed_time = end_time - start_time
    print_(result)
    print_("{:.3f}".format(elapsed_time))
    print_("{:.1f}".format(mem_usage))

if __name__ == "__main__":
    main()

File: src-tauri/implementations/rust/latest/factorial/wrapper.rs
mod factorial;
extern crate time;
extern crate sys_info;
use factorial::factorial;

fn main() {
    let start_time = time::precise_time_ns();
    let result = factorial(4);
    let end_time = time::precise_time_ns();
    let elapsed_time = (end_time - start_time) as f64 / 1_000_000_000.0;

    let mem_info = sys_info::mem_info().unwrap();
    let mem_usage = mem_info.total - mem_info.free;

    println!("{:?}", result);println!("{}", format!("{:.3}", elapsed_time));println!("{}", format!("{:.1}", mem_usage as f64 / 1024.0 / 1024.0));}